src/VirtoCommerce.NotificationsModule.Data/Services/NotificationMessageService.cs

```diff
@@ -85,9 +91,18 @@ private void ValidateMessageProperties(IEnumerable<NotificationMessage> messages
             }
 
             var validator = AbstractTypeFactory<NotificationMessageValidator>.TryCreateInstance();
-            foreach (var notification in messages)
+            foreach (var message in messages)
             {
-                validator.ValidateAndThrow(notification);
+                try
+                {
+                    validator.ValidateAndThrow(message);
```

Comment:
Better do not use try-catch for expected validation errors. Use `Validate` with return value instead.